### PR TITLE
Add CreateCallFailed, AnswerFailed events and StartInConferenceMode option

### PR DESCRIPTION
**Pull request by edwardho-ms**
- Add CreateCallFailed, AnswerFailed events 
- Add StartInConferenceMode option to CreateCall, CreateGroupCall and AnswerCall
- Regenerate autorest

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/communication/Azure.Communication.CallAutomation/src/CallAutomationClient.cs [hunk 1]
```diff
@@ -253,6 +253,7 @@ private AnswerCallRequestInternal CreateAnswerCallRequest(AnswerCallOptions opti
             request.TranscriptionConfiguration = CreateTranscriptionOptionsInternal(options.TranscriptionOptions);
             request.AnsweredBy = Source == null ? null : new CommunicationUserIdentifierModel(Source.Id);
             request.OperationContext = options.OperationContext;
+            request.StartInConferenceMode = options.StartInConferenceMode;
```
Comment (minwoolee-msft) [inline sdk/communication/Azure.Communication.CallAutomation/src/CallAutomationClient.cs:256]: Missing this in CreateCall route

File: sdk/communication/Azure.Communication.CallAutomation/src/Models/AnswerCallOptions.cs [hunk 2]
```diff
@@ -50,5 +50,10 @@ public AnswerCallOptions(string incomingCallContext, Uri callbackUri)
         /// The operation context.
         /// </summary>
         public string OperationContext { get; set; }
+
+        /// <summary>
+        /// If passed true, the call will start in an escalated state. (i.e. GroupCall).
+        /// </summary>
+        public bool? StartInConferenceMode { get; set; }
```
Comment (minwoolee-msft) [inline sdk/communication/Azure.Communication.CallAutomation/src/Models/AnswerCallOptions.cs:57]: is there reason to make this nullable?

File: sdk/communication/Azure.Communication.CallAutomation/src/CallAutomationEventProcessor/CallAutomationEventProcessor.cs [hunk 3]
```diff
@@ -62,8 +62,10 @@ public void ProcessEvents(IEnumerable<CallAutomationEventBase> events)
                     handlers(this, args);
                 }
 
-                // If this call is disconnect, remove all related items in memory
-                if (receivedEvent is CallDisconnected)
+                // If call is disconnected, CreateCall failed or Answer failed remove all related items in memory
+                if (receivedEvent is CallDisconnected
+                    || receivedEvent is CreateCallFailed
```
Comment (minwoolee-msft) [inline sdk/communication/Azure.Communication.CallAutomation/src/CallAutomationEventProcessor/CallAutomationEventProcessor.cs:67]: we should also add this in CreateCall / AnswerCall event processor result:
See here how it was implemented: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/communication/Azure.Communication.CallAutomation/src/CallAutomationEventProcessor/EventResult/CreateCallEventResult.cs

File: sdk/communication/Azure.Communication.CallAutomation/src/Models/AnswerCallOptions.cs (hunk 2 above)
Comment (edwardho-ms) [reply sdk/communication/Azure.Communication.CallAutomation/src/Models/AnswerCallOptions.cs:57]: added

**Pull request merged**